File: java/base/src/main/java/vsdk/toolkit/common/commandLineOptions/OptionGroupT.java
package vsdk.toolkit.common.commandLineOptions;

public class OptionGroupT<TOptionBase extends OptionBase> {
    public String name;
    public TOptionBase[] options;
    public int count;

    public OptionGroupT() {
        name = null;
        options = null;
        count = 0;
    }

    public OptionGroupT(String groupName, TOptionBase[] groupOptions, int groupCount) {
        name = groupName;
        options = groupOptions;
        count = groupCount;
    }
}

File: typescript/base/src/main/vsdk/toolkit/common/commandLineOptions/OptionGroupT.ts
import { OptionBase } from "./OptionBase";

export class OptionGroupT<TOptionBase extends OptionBase> {
  public name: string | null;
  public options: TOptionBase[] | null;
  public count: number;

  public constructor();
  public constructor(groupName: string, groupOptions: TOptionBase[], groupCount: number);
  public constructor(groupName?: string, groupOptions: TOptionBase[] | null = null, groupCount = 0) {
    this.name = groupName ?? null;
    this.options = groupOptions;
    this.count = groupCount;
  }
}
